components/backends/vllm/src/dynamo/vllm/handlers.py

```diff
@@ -161,8 +155,13 @@ async def generate(self, request):
                     "kv_transfer_params"
                 ] = prefill_response.kv_transfer_params
 
-        async for tok in self.generate_tokens(prompt, sampling_params, request_id):
-            yield tok
+        try:
+            async for tok in self.generate_tokens(prompt, sampling_params, request_id):
+                yield tok
+        except asyncio.CancelledError:
```

Comment:
@tedzhouhk do you think it is ok to move the try...catch from the inner `generate_tokens` method to the outer method here, for graceful engine shutdown?